relion/_parser/class2D.py

```diff
@@ -61,28 +61,13 @@ def __getitem__(self, key):
             self._jobcache[key] = self._load_job_directory(key)
         return self._jobcache[key]
 
-    def _find_values(self, value, data_or_model):
-        final_list = []
-        for x in self._basepath.iterdir():
-            if "job" in x.name:
-                job = x.name
-                val_list = []
-                if x.name not in self._jobcache:
-                    file = self.find_last_iteration(data_or_model)
-                    doc = self._read_star_file(job, file)
-                    val_list = list(self.parse_star_file(value, doc, 1))
-                final_list.append(val_list)
-        return final_list
-
     def _load_job_directory(self, jobdir):
-        # these are independent of jobdir, ie. this is a bug
-        dfile = self.find_last_iteration("data")
-        mfile = self.find_last_iteration("model")
-        # print(dfile, mfile)
+
+        dfile = self._final_data_and_model(self._basepath / jobdir)[0]
+        mfile = self._final_data_and_model(self._basepath / jobdir)[1]
```

Comment:
You can unpack the return value directly:
```suggestion
        dfile, mfile = self._final_data_and_model(self._basepath / jobdir)
```